src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -94,6 +96,17 @@ public boolean hasAnimal(Animal animal) {
         return zooKeepBook.hasAnimal(animal);
     }
 
+    @Override
+    public Optional<Animal> getAnimal(Id id) {
+        Animal found = null;
+        for (Animal animal : filteredAnimals) {
+            if (animal.getId().equals(id)) {
+                found = animal;
+            }
+        }
```

Comment:
Minor nitpick: The variable found sounds a bit vague, just a suggestion to put as foundAnimal or something more specific.